BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1420,136 +1420,143 @@ public void BeehiveDetonation()
                         break;
-                    case "AnalyticEstimate":
-                        pBullet.dragType = BulletDragTypes.AnalyticEstimate;
+                    case "proximity":
+                        sFuze = BulletFuzeTypes.Proximity;
                         break;
-                    case "NumericalIntegration":
-                        pBullet.dragType = BulletDragTypes.NumericalIntegration;
+                    case "flak":
+                        sFuze = BulletFuzeTypes.Flak;
+                        break;
+                    case "delay":
+                        sFuze = BulletFuzeTypes.Delay;
+                        break;
+                    case "penetrating":
+                        sFuze = BulletFuzeTypes.Penetrating;
+                        break;
+                    case "impact":
+                        sFuze = BulletFuzeTypes.Impact;
+                        break;
+                    case "none":
+                        sFuze = BulletFuzeTypes.Impact;
                         break;
                     default:
-                        pBullet.dragType = BulletDragTypes.AnalyticEstimate;
+                        sFuze = BulletFuzeTypes.None;
                         break;
                 }
-                pBullet.bullet = BulletInfo.bullets[subMunitionType.name];
-                pBullet.stealResources = stealResources;
-                pBullet.dmgMult = dmgMult;
-                pBullet.isAPSprojectile = isAPSprojectile;
-                pBullet.tgtShell = tgtShell;
-                pBullet.tgtRocket = tgtRocket;
-                pBullet.gameObject.SetActive(true);
-
-                if (pBullet.CheckBulletCollisions(iTime)) return; // Bullet immediately hit something.
-                pBullet.MoveBullet(iTime); // Move the bullet the remaining part of the frame.
-                pBullet.timeAlive = iTime;
+                if (BDArmorySettings.DEBUG_WEAPONS)
+                    Debug.Log("[BDArmory.PooledBullet]: Beehive Detonation: parsing submunition fuze: " + fuze + ", index: " + sFuze);
+                float incrementVelocity = 1000 / (bulletVelocity + sBullet.bulletVelocity); //using 1km/s as a reference Unit
+                float dispersionAngle = sBullet.subProjectileDispersion > 0 ? sBullet.subProjectileDispersion : BDAMath.Sqrt(count) / 2; //fewer fragments/pellets are going to be larger-> move slower, less dispersion
+                float dispersionVelocityforAngle = 1000 / incrementVelocity * Mathf.Sin(dispersionAngle * Mathf.Deg2Rad); // convert m/s despersion to angle, accounting for vel of round
+                for (int s = 0; s < count; s++) //this does mean that setting a subMunitionType to, say, shotgun shells and then setting a sMT projectile count of, say, 5, would have only 5 shotgun pellets spawn, even if the shutgun shell projectileCount = 30. Could always have it be count * subMunitiontype.projectileCount if you want shotshells as an allowable submunition
```

Comment:
It doesn't really make sense, no, since if you can specify the number of submunitions in the parent shell, just setting it to 150 in this example is much more straightforward/intuitive. but it would cover this if there are any mod beehive ammos out there that need changing over.
For recursive ammo, it would still be ``s < count; s++`` with modifications to TimeToDetonation and detonationRange, but it does seem a bit superfluous, yes - after, why not just have more submunitions in the single shell/wider dispersion angle to begin with?